cocos2d/core/platform/CCTypes.js

```diff
@@ -794,7 +794,7 @@ cc.hexToColor = function (hex) {
     var r = c >> 16;
     var g = (c >> 8) % 256;
     var b = c % 256;
-    return new cc.Color(r, g, b);
```

Comment:
I agree with you, can you modify the cc.Color constructor instead ?

```
a = typeof a === 'number' ? a : 255;
```

It would be cleaner.